pkg/dispatcher/dispatcher.go

```diff
@@ -169,20 +172,28 @@ func (dispatcher *Dispatcher) dispatch(ssn *dispatcherframework.Session) {
 		}
 
 		queue := queues.Pop().(*schedulingapi.QueueInfo)
-		resourceBindingsQueue := resourceBindingMap[queue.Name]
-
-		// Get all the ResourceBindingInfos from the priority queue.
-		for !resourceBindingsQueue.Empty() {
-			rbi := resourceBindingsQueue.Pop().(*api.ResourceBindingInfo)
-
-			rbi.DispatchStatus = api.UnSuspending
-			dispatcher.cache.UnSuspendResourceBinding(types.NamespacedName{
-				Namespace: rbi.ResourceBinding.Namespace,
-				Name:      rbi.ResourceBinding.Name,
-			})
-			dispatchResourceBindingCount++
+
+		resourceBindings, found := resourceBindingMap[queue.Name]
+		if !found || resourceBindings.Empty() {
+			// This queue has no ResourceBinding, skip it.
+			continue
+		}
+
+		// Get the highest priority ResourceBinding from the queue.
+		rbi := resourceBindings.Pop().(*api.ResourceBindingInfo)
+
+		rbi.DispatchStatus = api.UnSuspending
+		dispatcher.cache.UnSuspendResourceBinding(types.NamespacedName{
+			Namespace: rbi.ResourceBinding.Namespace,
+			Name:      rbi.ResourceBinding.Name,
+		})
+		dispatchResourceBindingCount++
+
+		// Re-add the queue if it still has ResourceBindings.
+		if !resourceBindings.Empty() {
```

Comment:
If the queue has higher priority, it will still be processed next round: )

Reply:
I'm still uncertain about this🤔. Here are some options:

1. Modify queue sorting - but this breaks Queue Priority semantics
2. Don't re-enqueue after dequeue - ensures fairness but hurts performance badly
3. Add Volcano-style `Overused` - seems promising and worth exploring next

Reply:
Option 3 is a good way,  but currently we have not implemented this mechanism: )